transmission_interface/include/transmission_interface/simple_transmission.hpp

```diff
@@ -0,0 +1,240 @@
+// limitations under the License.
+
+#ifndef TRANSMISSION_INTERFACE__SIMPLE_TRANSMISSION_HPP_
+#define TRANSMISSION_INTERFACE__SIMPLE_TRANSMISSION_HPP_
+
+#include <algorithm>
+#include <cassert>
+#include <string>
+#include <vector>
+
+#include "transmission_interface/transmission.hpp"
+#include "transmission_interface/exception.hpp"
+
+namespace transmission_interface
+{
+
+/**
+ * \brief Implementation of a simple reducer transmission.
+ *
+ * This transmission relates <b>one actuator</b> and <b>one joint</b> through a reductor (or amplifier).
+ * Timing belts and gears are examples of this transmission type, and are illustrated below.
+ * \image html simple_transmission.png
+ *
+ * <CENTER>
+ * <table>
+ * <tr><th></th><th><CENTER>Effort</CENTER></th><th><CENTER>Velocity</CENTER></th><th><CENTER>Position</CENTER></th></tr>
+ * <tr><td>
+ * <b> Actuator to joint </b>
+ * </td>
+ * <td>
+ * \f[ \tau_j = n \tau_a \f]
+ * </td>
+ * <td>
+ * \f[ \dot{x}_j = \dot{x}_a / n \f]
+ * </td>
+ * <td>
+ * \f[ x_j = x_a / n + x_{off} \f]
+ * </td>
+ * </tr>
+ * <tr><td>
+ * <b> Joint to actuator </b>
+ * </td>
+ * <td>
+ * \f[ \tau_a = \tau_j / n\f]
+ * </td>
+ * <td>
+ * \f[ \dot{x}_a = n \dot{x}_j \f]
+ * </td>
+ * <td>
+ * \f[ x_a = n (x_j - x_{off}) \f]
+ * </td></tr></table>
+ * </CENTER>
+ *
+ * where:
+ * - \f$ x \f$, \f$ \dot{x} \f$ and \f$ \tau \f$ are position, velocity and effort variables, respectively.
+ * - Subindices \f$ _a \f$ and \f$ _j \f$ are used to represent actuator-space and joint-space variables, respectively.
+ * - \f$ x_{off}\f$ represents the offset between motor and joint zeros, expressed in joint position coordinates.
+ * - \f$ n \f$ is the transmission ratio, and can be computed as the ratio between the output and input pulley
+ *   radii for the timing belt; or the ratio between output and input teeth for the gear system.
+ *   The transmission ratio can take any real value \e except zero. In particular:
+ *     - If its absolute value is greater than one, it's a velocity reducer / effort amplifier, while if its absolute
+ *       value lies in \f$ (0, 1) \f$ it's a velocity amplifier / effort reducer.
+ *     - Negative values represent a direction flip, ie. actuator and joint move in opposite directions. For example,
+ *       in timing belts actuator and joint move in the same direction, while in single-stage gear systems actuator and
+ *       joint move in opposite directions.
+ *
+ * \ingroup transmission_types
+ */
+class SimpleTransmission : public Transmission
+{
+public:
+  /**
+   * \param reduction Reduction ratio.
+   * \param joint_offset Joint position offset used in the position mappings.
+   * \pre Nonzero reduction value.
+   */
+  SimpleTransmission(
+    const double reduction,
+    const double joint_offset = 0.0);
+
+  void configure(
+    const std::vector<JointHandle> & joint_handles,
+    const std::vector<ActuatorHandle> & actuator_handles) override;
```

Comment:
We should document that it is expected that you can provide multiple Handles but only of  different interfaces of the same joint. I did not understand it until I got to the implementation of this method.

Reply:
Can we check this somehow to protect users from making errors?

Reply:
@destogl it is checked and exceptions are thrown accurately reporting the issue. I'll add docs to reinforce